google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java

```diff
@@ -540,29 +540,45 @@ public Row readRow(String tableId, ByteString rowKey, @Nullable Filter filter) {
    * <p>Sample code:
    *
    * <pre>{@code
+   * class ClientObserver implements ResponseObserver<Row> {
```

Comment:
Please keep this as an anonymous inner class, something like:

```java

   *   bigtableDataClient.readRowsAsync(query, new ResponseObserver<Row>() {
   *     StreamController controller;
   *     int count = 0;
   *
   *     public void onStart(StreamController controller) {
   *         this.controller = controller;
   *     }
   *     public void onResponse(Row row) {
   *        if (++count > 10) {
  *            controller.cancel();
   *           return;
  *        }
   *       // Do something with Row
   *     }
   *     public void onError(Throwable t) {
   *       if (t instanceof NotFoundException) {
   *         System.out.println("Tried to read a non-existent table");
   *       } else {
   *         t.printStackTrace();
   *       }
   *     }
   *     public void onComplete() {
   *       // Handle stream completion
   *     }
   *   });

```

Reply:
Have updated the sample code to include the described approach. Please have a fresh look